types/errors/abci.go

```diff
@@ -95,8 +95,8 @@ func ResponseDeliverTxWithEvents(err error, gw, gu uint64, events []abci.Event,
 
 // QueryResult returns a ResponseQuery from an error. It will try to parse ABCI
 // info from the error.
-func QueryResult(err error) abci.ResponseQuery {
-	space, code, log := ABCIInfo(err, false)
+func QueryResult(err error, debug bool) abci.ResponseQuery {
```

Comment:
We cannot backport this function, because it's API-breaking.

What we can do is create a new function `QueryResultWithDebug`, does that work for you @yihuang ?

Reply:
ACK -- create a `QueryResultWithDebug` 👍